File: driver-service/src/test/java/com/software/modsen/driverservice/integration/PostgresTestContainerSetup.java
package com.software.modsen.driverservice.integration;

import org.junit.jupiter.api.BeforeAll;
import org.testcontainers.containers.PostgreSQLContainer;

public class PostgresTestContainerSetup {
    private static PostgreSQLContainer<?> postgreSQLContainer;

    @BeforeAll
    public static void setUpPostgresDB() {
        postgreSQLContainer = new PostgreSQLContainer<>("postgres:15-alpine")
                .withDatabaseName("test_db")
                .withUsername("postgres")
                .withPassword("root");
        postgreSQLContainer.start();

        System.setProperty("spring.datasource.url", postgreSQLContainer.getJdbcUrl());
        System.setProperty("spring.datasource.username", postgreSQLContainer.getUsername());
        System.setProperty("spring.datasource.password", postgreSQLContainer.getPassword());
    }
}

File: passenger-service/src/test/kotlin/com/software/modsen/passengerservice/integration/PostgresTestContainerSetup.kt
package com.software.modsen.passengerservice.integration

import org.junit.jupiter.api.BeforeAll
import org.testcontainers.containers.PostgreSQLContainer

open class PostgresTestContainerSetup {
    companion object {
        private lateinit var postgreSQLContainer: PostgreSQLContainer<*>

        @JvmStatic
        @BeforeAll
        fun setUpPostgresDB() {
            postgreSQLContainer = PostgreSQLContainer<Nothing>("postgres:15-alpine").apply {
                withDatabaseName("test_db")
                withUsername("postgres")
                withPassword("root")
            }
            postgreSQLContainer.start()

            System.setProperty("spring.datasource.url", postgreSQLContainer.jdbcUrl)
            System.setProperty("spring.datasource.username", postgreSQLContainer.username)
            System.setProperty("spring.datasource.password", postgreSQLContainer.password)
        }
    }
}